distributed/worker.py

```diff
@@ -1475,7 +1475,10 @@ async def close(
         # nanny+worker, the nanny must be notified first. ==> Remove kwarg
         # nanny, see also Scheduler.retire_workers
         if self.status in (Status.closed, Status.closing, Status.failed):
-            await self.finished()
+            try:
+                await self.finished()
+            except asyncio.CancelledError:
+                pass
```

Comment:
can you add a comment why this is necessary? Just swallowing a CancelledError feels wrong. why should this not bubble up?

Reply:
Good catch, this has been a leftover from an issue that has been fixed differently since.

Reply:
Good catch, this is a leftover of an issue that has been solved differently since then.